Added detailed error code explanation for issue E0688 in Rust compiler. (#72654)
rust-lang/rust: src/librustc_error_codes/error_codes/E0688.md

> In-band lifetimes has mixed with explicit lifetime binders.

Reviewer: ```suggestion
In-band lifetimes were mixed with explicit lifetime binders.
```

Erroneous code example:

```compile_fail,E0688
#![feature(in_band_lifetimes)]

fn foo<'a>(x: &'a u32, y: &'b u32) {}   // error!

struct Foo<'a> { x: &'a u32 }

impl Foo<'a> {
    fn bar<'b>(x: &'a u32, y: &'b u32, z: &'c u32) {}   // error!
}

impl<'b> Foo<'a> {  // error!
    fn baz() {}
}
```

> We cannot mix in-band lifetimes with explicit lifetime binders.

Reviewer: ```suggestion
In-band lifetimes cannot be mixed with explicit lifetime binders.
```

This is really just a re-phrasing of the error, it'd be great to elaborate a little - something like "a function/struct/whatever can either have explicit lifetime binders *or* in-band lifetimes".

For example:

```

> fn foo<'a, 'b>(x: &'a u32, y: &'b u32) {}   // ok!

Reviewer: I'd also add another example where only in-band lifetimes are used (with a short sentence before each example explaining how they differ).
